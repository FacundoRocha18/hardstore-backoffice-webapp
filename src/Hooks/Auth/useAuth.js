import { useState } from 'react';

/* Custom hooks --------------------- */
import { useToken } from './useToken';

/* API Services --------------------- */
import { LoginService } from '../../Services';

export const useAuth = () => {

  const [loading, setIsLoading] = useState(true)

  const { token, setToken } = useToken()

  const handleLogin = async ({ id, pin }) => {

    try {
      const {
        status,
        message,
        token
      } = await LoginService(id, pin);

      if (status === false) {
        return console.log(message)
      }

      setTimeout(() => {
        return {
          token: setToken(token),
        };
      }, 2000);

      return loading;
     
    } catch (error) {
      console.log(error)
    }
  }

  const handleLogout = () => {
    sessionStorage.clear();
    window.location.reload(false);
  };

  return {
    token,
    loading,
    setIsLoading,
    onLogin: handleLogin,
    onLogOut: handleLogout
  }
}


