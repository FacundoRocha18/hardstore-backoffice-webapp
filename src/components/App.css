.App {
  text-align: center;
}

@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@100;200;300;400;500;600;700;800;900&display=swap');

html {
    box-sizing: border-box;
    font-size: 62.5%;
    /** 1rem = 10px**/
    padding: 0;
    margin: 0;
    max-height: 100%;
    max-width: 100%;
    scroll-behavior: smooth;
    overflow-x: hidden;
}

* {
    padding: 0;
    margin: 0;
    box-sizing: border-box;
}

body {
  font-family: 'Poppins', sans-serif;
  background-color: #0D0D0D;   
  width: 100%;
  height: 100%;
  color: #fff;
  font-size: 1.6rem;
  padding: 0;
  margin: 1px;
}


h1, h2, h3, h4, h5, h6 {
  font-family: 'Poppins', sans-serif;
  color: var(--headers-main-color);
}

h1 {
  font-size: 3.9rem;
  font-weight: var(--semi-bold);
}

h2 {
  font-size: 3.1rem;
  font-weight: var(--semi-bold);
}

h3 {
  font-size: 2.5rem;
  font-weight: var(--medium-h);
}

h4 {
  font-size: 2.2rem;
  font-weight: var(--medium);
}

h5 {
  font-size: 2rem;
  font-weight: var(--medium);
}

h6 {
  font-size: 1.8rem;
  font-weight: var(--medium);
}

a {
  text-decoration: none;
  color: var(--text-main-color);
}

p {
  font-family: 'Poppins', sans-serif;
  font-size: 1.6rem;
  font-weight: var(--medium);
}

ul, ol, li {
  list-style: none;
}

:root {

  /* Colors */
  --blue: #039BE5;
  --blue-hover: #0277BD;
  --orange: #ff6f00dc;
  --orange-hover: #c25705;
  --black: #0d0d0d;
  --black-hover: #292929;
  --white: #ffffff;

  /* General */
  --body-bg-main-color: #ffffff;
  --item-bg-main-color: #d3d3d3;
  --item-bg-main-color-blur: #d3d3d341;
  --item-bg-secondary-color: #0d0d0d;
  --product-item-background: #dddcdca1;

  /* Buttons */
  --primary-btn-bg-color: #ff6d00;
  --outline-btn-bg-color: #c4c5cd;
  --success-btn-bg-color: #66BB6A;
  --critical-btn-bg-color: #ec4644;
  --login-btn: #039BE5;

  /* Buttons hover states */
  --primary-btn-hover-bg-color: #c25705;
  --success-btn-hover-bg-color: #58a55c;
  --critical-btn-hover-bg-color: #c43230;
  --login-btn-hover: #0277BD;

  /* Borders */
  --borders-main-color: #151515;
  --borders-main-color-blur: #494949;
  --border-grey-color: #6e6e6e;


  /* Fonts */
  --headers-main-color: #0d0d0d;
  --text-main-color: #111318;
  --text-secondary-color: #ff6d00;
  --text-white: #dad6d6;

  /* Gradient backgrounds */
  --site-background-gradient: linear-gradient(156deg, rgba(0,0,0,1) 30%, rgba(32,32,32,1) 80%); 
  --primary-btn-gradient: linear-gradient(190deg, rgba(255, 111, 0, 1) 20%, rgba(244, 81, 30, 1) 90%);
  --primary-btn-gradient-hover: linear-gradient(190deg, rgb(255, 126, 28) 20%, rgb(247, 100, 55) 90%);
  --success-btn-gradient: linear-gradient(190deg, rgba(102, 187, 106, 1) 20%, rgba(56, 142, 60, 1) 90%);
  --success-btn-gradient-hover: linear-gradient(190deg, rgba(113, 202, 117, 1) 20%, rgba(70, 165, 75, 1) 90%);
  --critical-btn-gradient: linear-gradient(190deg, rgba(236, 70, 68, 1) 20%, rgba(196, 50, 48, 1) 90%);
  --critical-btn-gradient-hover: linear-gradient(190deg, rgba(247, 89, 87, 1) 20%, rgba(222, 66, 64, 1) 90%);


  /* Box shadow */
  --box-shadow-1: 0px 0px 15px 0px rgba(0, 0, 0, 0.1);
  --box-shadow-2: 0px 0px 15px 5px rgba(0, 0, 0, 0.2);


  /* Font weight */
  --extra-light: 200;
  --light: 300;
  --medium: 400;
  --medium-h: 500;
  --semi-bold: 600;
  --bold: 700;
}

/* Buttons */

.btn {
  border: none;
  border-radius: .5rem;
  padding: 1rem 2rem;
  transition: all .5s ease;
}

.main_container {
  margin: 0 auto;
  width: 90%;
}

.website_logo {
  color: var(--text-secondary-color);
  font-weight: var(--extra-light);
  letter-spacing: -1.5pt;
  margin: 1.5rem auto;
  text-align: left;
}

.website_logo span {
  color: var(--text-white);
  font-weight: var(--medium-h);
  letter-spacing: -1pt;
}

.website_logo p {
  font-size: 2rem;
}

/* Buttons */

.icon_btn {
  background-color: transparent;
  border: none;
  border-radius: 1rem;
  cursor: pointer;
  padding: .5rem;
  max-width: 45px;
  max-height: 45px;
  transition: all .5s ease;
  z-index: 1000;
}

.icon_btn span {
  font-size: 3.5rem;
}

.icon_btn:hover {
  background-color: var(--orange);
}


