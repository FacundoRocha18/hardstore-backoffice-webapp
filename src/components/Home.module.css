.home_container {
    width: calc(100% - 375px);
    height: 100%;
    position: absolute;
    right: 0;
    top: 0;
    transition: all .5s ease;
}

.home_container.grow {
    width: calc(100% - 75px);
}