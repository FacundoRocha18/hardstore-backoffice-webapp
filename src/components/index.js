export * from './App'
export * from './Home'
export * from './login/Login'
export * from './navigation/Header'
export * from './navigation/Menu'
export * from './screens/OptionsScreen'
export * from './screens/LoadingScreen';



