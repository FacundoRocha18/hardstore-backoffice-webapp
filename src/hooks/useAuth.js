import { useState } from 'react';

/* Custom hooks --------------------- */
import { useToken } from '.';

/* API getters --------------------- */
import { getAuth } from '../API';

export const useAuth = () => {

  const [loading, setIsLoading] = useState(true)

  const { token, setToken } = useToken()

  const handleLogin = async ({ id, pin }) => {

    try {
      const {
        status,
        message,
        token
      } = await getAuth(id, pin);

      if (status === false) {
        return console.log(message)
      }

      setTimeout(() => {
        return {
          token: setToken(token),
        };
      }, 2000);

      return loading;
     
    } catch (error) {
      console.log(error)
    }
  }

  const handleLogout = () => {
    sessionStorage.clear();
    window.location.reload(false);
  };

  return {
    token,
    loading,
    onLogin: handleLogin,
    onLogOut: handleLogout
  }
}


