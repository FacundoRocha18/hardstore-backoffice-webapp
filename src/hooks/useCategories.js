import { useEffect, useState } from 'react'

import { getCategories } from '../API'


export const useCategories = () => {

    const [cats, setCats] = useState([])

    useEffect(() => {
        getCats();

    }, [])


    const getCats = async () => {

        const request = await getCategories();

        setCats(request);
    }

    return {
        cats
    }
}