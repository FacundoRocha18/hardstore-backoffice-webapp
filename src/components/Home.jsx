import React from 'react'

/* CSS Styles --------------------- */
import style from './Home.module.css'
import css from 'classnames'

/* Components ----------------------- */
import { Header, LoadingScreen } from '../components'

export const Home = ({showMenu, loading}) => {
  
  if ( loading ) {
    return (
      <LoadingScreen />
    )
  }

  if ( !loading ) {
    return (
      <>
          <div className={css(style.home_container, !showMenu && style.grow)}>
          </div>
      </>
    )
  }

}

