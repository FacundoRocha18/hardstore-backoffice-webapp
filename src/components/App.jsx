import React, { useState } from 'react';

/* ReactRouter ----------------------- */
import {
  BrowserRouter as Router,
  Route,
  Routes,
  useNavigate
} from 'react-router-dom'

/* Components ----------------------- */
import { Home, Header, LoginScreen } from '../components'

/* Custom hooks --------------------- */
import { useAuth } from '../hooks'

/* CSS Styles ----------------------- */
import './App.css';

export const App = () => {

  const [showMenu, setShowMenu] = useState(true);

  const { token, loading, setIsLoading, onLogin, onLogOut } = useAuth()

  if (!token) {
    return <LoginScreen onLogin={onLogin} />
  }

  return (
    <div className="App">
      <Router>
        <Header showMenu={showMenu} setShowMenu={setShowMenu} onLogOut={onLogOut} />
        <Routes>
          <Route
            path='/'
            element={
              <Home showMenu={showMenu} isLoading={loading} setIsLoading={setIsLoading} />
            } />
        </Routes>
      </Router>
    </div>
  );
}

