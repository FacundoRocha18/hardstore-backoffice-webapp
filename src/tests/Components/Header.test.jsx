import { Header } from '../../components';
import { render, screen } from '@testing-library/react';
import { useAuth } from '../../hooks/useAuth';

describe('Tests on <Header /> component', () => {

    const { onLogOut } = useAuth();

    test('should match the snapshot', () => {

        const { container } = render(
            <Header
                onLogOut={onLogOut}
            />
        )

        expect(container).toMatchSnapshot();
    })
})