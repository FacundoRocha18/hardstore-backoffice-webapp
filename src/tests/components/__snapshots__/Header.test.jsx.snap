// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <Header /> component should match the snapshot 1`] = `
<div>
  <header
    class="hidden"
  >
    <div
      class="header_container hide"
    >
      <section
        class="header_content"
      >
        <div
          class="header_title_container"
        >
          <h1
            class="website_logo"
          >
            <span>
              Hard
            </span>
            Store
          </h1>
        </div>
      </section>
      <section
        class="header_menu hidden"
      >
        <div
          class="menu_container"
        >
          <div
            class="menu_header"
          >
            <h3>
              Menú
            </h3>
          </div>
          <nav
            class="menu"
          >
            <ul>
              <li>
                <a
                  href="/"
                >
                  Inicio
                </a>
                <span
                  class="material-icons-round"
                >
                  home
                </span>
              </li>
              <li>
                <a
                  href="/"
                >
                  Empleados
                </a>
                <span
                  class="material-icons-round"
                >
                  people_alt
                </span>
              </li>
              <li>
                <a
                  href="/"
                >
                  Inventario
                </a>
                <span
                  class="material-icons-round"
                >
                  inventory
                </span>
              </li>
              <li>
                <a
                  href="/"
                >
                  Ventas
                </a>
                <span
                  class="material-icons-round"
                >
                  sell
                </span>
              </li>
            </ul>
          </nav>
        </div>
      </section>
    </div>
    <div
      class="buttons_container move"
    >
      <button
        class="icon_btn toggle_button"
        title="Menu"
      >
        <span
          class="material-icons-round"
        >
          close
        </span>
      </button>
      <section
        class="logout_button_container hidden"
      >
        <button
          class="icon_btn logout_button"
          title="Cerrar sesión"
        >
          <span
            class="material-icons-round"
          >
            logout
          </span>
        </button>
      </section>
    </div>
  </header>
</div>
`;
